### DISCUSSION TITLE
LARGE REPO FEATURE REQUEST: Allow merge conflict resolution in the UI when the conflict happens within the first (20/100/500) lines of the file

### DISCUSSION THREAD

**Discussion by corsonknowles** [community/community#127447] [Enterprise]
### Body

TL:DR manually resolving merge conflicts on Rails migrations is tedious when the only change to reconcile is the schema date. In large apps, or apps with large table structure, this soon exceeds the max file size that Github currently supports for resolving conflicts in the UI. 

PROPOSED FIX: Resolving conflicts in the Github UI is faster and more pleasant than switching to the cli/editor. Instead of checking file size to decide if conflicts can be resolved within Github, check that the conflict appears within the first N lines of the file (ideally the same number of lines that Github supports for files of that size).

This will provide a huge quality of life increase for Rails engineers (like the talented Github team) and other major applications in the Ruby space. And this will have a broader and beneficial impact across all large company users and many open source maintainers, often anywhere that generated files need some heading changes during merge.

**Comment by tilo** [comment]
e.g. schema file with +30,000 loc
